src/cli_plugin/install/zip.js

```
@@ -1,98 +1,120 @@
-  return function filterByRootPath(file) {
-    if (file.type === SYMBOLIC_LINK) return false;
+export function analyzeArchive(archive) {
+  const plugins = [];
+  const regExp = new RegExp('(kibana[\\\\/][^\\\\/]+)[\\\\/]package\.json', 'i');
 
-    return paths.some(path => {
-      const regex = new RegExp(`${path}($|/)`, 'i');
-      return file.parent.match(regex);
-    });
-  };
-}
+  return new Promise ((resolve, reject) => {
+    yauzl.open(archive, { lazyEntries: true }, function (err, zipfile) {
+      if (err) {
+        return reject(`unable to read ${archive}`);
+      }
 
-/**
- * Creates a filter function to be consumed by extractFiles
- * @param {object} filter - an object with either a files or paths property.
- */
-function extractFilter(filter) {
-  if (filter.files) return extractFilterFromFiles(filter.files);
-  if (filter.paths) return extractFilterFromPaths(filter.paths);
-  return _.noop;
-}
+      zipfile.readEntry();
+      zipfile.on('entry', function (entry) {
+        const match = entry.fileName.match(regExp);
 
-/**
- * Extracts files from a zip archive to a file path using a filter function
- * @param {string} zipPath - file path to a zip archive
- * @param {string} targetPath - directory path to where the files should
- *  extracted
- * @param {integer} strip - Number of nested directories within the archive
- *  that should be ignored when determining the target path of an archived
- *  file.
- * @param {function} filter - A function that accepts a single parameter 'file'
- *  and returns true if the file should be extracted from the archive
- */
-export async function extractFiles(zipPath, targetPath, strip, filter) {
-  await new Promise((resolve, reject) => {
-    const unzipper = new DecompressZip(zipPath);
+        if (match) {
+          zipfile.openReadStream(entry, function (err, readable) {
+            const chunks = [];
+
+            if (err) {
+              return reject(`unable to read ${entry.fileName} within zip`);
```

Same as above re: swallowing errors.